File: RestReactAspire.Server/Models/Patient.cs
namespace RestReactAspire.Server.Models;

public class Patient
{
    public Guid Id { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public DateOnly DateOfBirth { get; set; }
    public required string Email { get; set; }
    public required string Phone { get; set; }
}

File: frontend/src/types/patient.ts
import type { Link, PaginationInfo } from './hateoas';

export interface Patient {
  id: string;
  firstName: string;
  lastName: string;
  dateOfBirth: string;
  email: string;
  phone: string;
  links: Link[];
}

export interface PatientList {
  items: Patient[];
  pagination: PaginationInfo;
  links: Link[];
}

export interface CreatePatientRequest {
  firstName: string;
  lastName: string;
  dateOfBirth: string;
  email: string;
  phone: string;
}

export interface UpdatePatientRequest {
  firstName: string;
  lastName: string;
  dateOfBirth: string;
  email: string;
  phone: string;
}
